vembrane/__init__.py

```diff
@@ -48,42 +52,55 @@
 }
 
 
-class Sample:
-    def __init__(self, record_idx: int, sample_name: str, sample: VariantRecordSample):
+class Format:
+    def __init__(
+        self, record_idx: int, name: str, record_samples: VariantRecordSamples
+    ):
         self._record_idx = record_idx
-        self._sample_name = sample_name
-        self._sample = sample
-        self._data = {}
+        self._name = name
+        self._record_samples = record_samples
+        self._sample_values = {}
 
-    def __getitem__(self, item):
+    def __getitem__(self, sample):
         try:
-            return self._data[item]
+            return self._sample_values[sample]
         except KeyError:
             try:
-                sample_format = self._sample[item]
+                record_sample = self._record_samples[sample]
             except KeyError as ke:
-                raise UnknownFormatField(self._record_idx, self._sample_name, ke)
-            self._data[item] = sample_format
-            return sample_format
+                raise UnknownSample(
+                    self._record_idx, str(ke).split(":")[1][1:].strip("'")
```

Comment:
What does the `str(ke)` contain and what are you extracting? Something more than just `sample`?

Reply:
Yes, this is ugly, but the KeyError's sole arg is `'invalid sample name: SomeUnknownSampleName'`. Oh wow I could have just used `sample` here, nvm, will fix.